src/socket/raw.rs

```diff
@@ -291,123 +302,238 @@ mod test {
+        #[cfg(feature = "proto-ipv4")]
+        {
+            let mut socket = ipv4_locals::socket(buffer(1), buffer(0));
+            assert!(!socket.can_recv());
+
+            let mut cksumd_packet = ipv4_locals::PACKET_BYTES.clone();
+            Ipv4Packet::new(&mut cksumd_packet).fill_checksum();
+
+            assert_eq!(socket.recv(), Err(Error::Exhausted));
+            assert!(socket.accepts(&ipv4_locals::HEADER_REPR));
+            assert_eq!(socket.process(&ipv4_locals::HEADER_REPR, &ipv4_locals::PACKET_PAYLOAD,
+                                      &ChecksumCapabilities::default()),
+                       Ok(()));
+            assert!(socket.can_recv());
+
+            assert!(socket.accepts(&ipv4_locals::HEADER_REPR));
+            assert_eq!(socket.process(&ipv4_locals::HEADER_REPR, &ipv4_locals::PACKET_PAYLOAD,
+                                      &ChecksumCapabilities::default()),
+                       Err(Error::Exhausted));
+            assert_eq!(socket.recv(), Ok(&cksumd_packet[..]));
+            assert!(!socket.can_recv());
+        }
+        #[cfg(feature = "proto-ipv6")]
+        {
+            let mut socket = ipv6_locals::socket(buffer(1), buffer(0));
+            assert!(!socket.can_recv());
+
+            assert_eq!(socket.recv(), Err(Error::Exhausted));
+            assert!(socket.accepts(&ipv6_locals::HEADER_REPR));
+            assert_eq!(socket.process(&ipv6_locals::HEADER_REPR, &ipv6_locals::PACKET_PAYLOAD,
+                                      &ChecksumCapabilities::default()),
+                       Ok(()));
+            assert!(socket.can_recv());
+
+            assert!(socket.accepts(&ipv6_locals::HEADER_REPR));
+            assert_eq!(socket.process(&ipv6_locals::HEADER_REPR, &ipv6_locals::PACKET_PAYLOAD,
+                                      &ChecksumCapabilities::default()),
+                       Err(Error::Exhausted));
+            assert_eq!(socket.recv(), Ok(&ipv6_locals::PACKET_BYTES[..]));
+            assert!(!socket.can_recv());
+        }
     }
 
     #[test]
-    #[cfg(feature = "proto-ipv4")]
     fn test_doesnt_accept_wrong_proto() {
-        let socket = match RawSocket::new(IpVersion::Ipv4,
-                                          IpProtocol::Unknown(ipv4_locals::IP_PROTO+1),
-                                          buffer(1), buffer(1)) {
-            Socket::Raw(socket) => socket,
-            _ => unreachable!()
-        };
-        assert!(!socket.accepts(&ipv4_locals::HEADER_REPR));
+        #[cfg(feature = "proto-ipv4")]
+        {
+            let socket = match RawSocket::new(IpVersion::Ipv4,
+                                              IpProtocol::Unknown(ipv4_locals::IP_PROTO+1),
+                                              buffer(1), buffer(1)) {
+                Socket::Raw(socket) => socket,
+                _ => unreachable!()
+            };
+            assert!(!socket.accepts(&ipv4_locals::HEADER_REPR));
+            #[cfg(feature = "proto-ipv6")]
```

Comment:
Useless `#[cfg]`.

Reply:
I use an ipv6 specific variable below and this block is run if `proto-ipv4` is enabled, so if `proto-ipv4` is enabled but `proto-ipv6` is not you'll get a compile failure.

Reply:
Oh, I misread.